lib/active_job/retry.rb

```diff
@@ -117,6 +125,18 @@ def check_adapter!
       end
     end
 
+    def validate_params
+      if retry_callback && !retry_callback.is_a?(Proc)
```

Comment:
Well, basically you can access everything you have defined for you current job instance, for example instance variables. Or methods of that instance (in particular the methods added by activejob-retry too, for example `serialize`). I currently use it like this:

    include ActiveJob::Retry.new(
      ...
      callback: proc do |exception, delay|
        Rails.logger.warn(
          "Retrying because of exception \"#{exception.message}\" in #{delay} seconds, retry attempt #{retry_attempt}"
        )
        ExceptionNotifier.notify_exception(
          exception, data: { delay: delay }.merge(serialize)
        )
      end
    )